File: sdks/php/src/Models/Notification.php
<?php

namespace Puxbay\Models;

class Notification
{
    public function __construct(
        public readonly string $id,
        public readonly string $title,
        public readonly string $message,
        public readonly bool $isRead,
        public readonly string $notificationType,
        public readonly \DateTimeImmutable $createdAt,
    ) {
    }

    public static function fromArray(array $data): self
    {
        return new self(
            id: $data['id'],
            title: $data['title'],
            message: $data['message'],
            isRead: $data['is_read'],
            notificationType: $data['notification_type'],
            createdAt: new \DateTimeImmutable($data['created_at']),
        );
    }

    public function toArray(): array
    {
        return [
            'id' => $this->id,
            'title' => $this->title,
            'message' => $this->message,
            'is_read' => $this->isRead,
            'notification_type' => $this->notificationType,
            'created_at' => $this->createdAt->format(\DateTimeInterface::ATOM),
        ];
    }
}

File: sdks/dotnet/src/Models/Notification.cs
using System;
using Newtonsoft.Json;

namespace Puxbay.SDK.Models
{
    public class Notification
    {
        [JsonProperty("id")]
        public string Id { get; set; }

        [JsonProperty("title")]
        public string Title { get; set; }

        [JsonProperty("message")]
        public string Message { get; set; }

        [JsonProperty("is_read")]
        public bool IsRead { get; set; }

        [JsonProperty("notification_type")]
        public string NotificationType { get; set; }

        [JsonProperty("created_at")]
        public DateTime? CreatedAt { get; set; }
    }
}
